drud/ddev
# Rework pantheon integration to use terminus inside web container, fixes #1731, fixes #1232

## The Problem/Issue/Bug:

#1731: The Pantheon API is changing (and remains undocumented), so we'll use terminus instead.
#1232: Ddev config didn't seem to work in particularly large environments. Test to see if this solves it since we're using terminus directly.

## How this PR Solves The Problem:

* Include terminus inside the web container
* Since terminus isn't supported on Windows, that solves the windows problem
* `ddev pull` can use terminus with an exec to accomplish what's needed

- [x] `ddev auth-pantheon` should become `ddev auth pantheon`
- [x] Update docs for all usages, but change from auth-pantheon to `auth pantheon`

## Manual Testing Instructions:

* Read the [(new) docs](https://github.com/drud/ddev/blob/ba93db069c7d17f509ee3eb53160c6ecd6c4dbcd/docs/users/providers/pantheon.md), as `ddev auth-pantheon` becomes `ddev auth pantheon`. 
As in the standard instructions, you need a pantheon site to work with, and get a token there.
* `ddev auth pantheon <token>`
* `ddev config pantheon`
* `ddev pull`

* Return to #1232 and see of one of those folks can test to see if that problem is resolved.

## Automated Testing Overview:

The original tests are still there.
One previously commented-out test has been added back in.

## Related Issue Link(s):

OP #1731 

## Release/Deployment notes:

* ddev auth-pantheon becomes ddev auth pantheon
